src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1425,6 +1449,121 @@ data LockingKind
       --
       -- @since 2.2.7
 
+newtype (ComparableLockStrength a) => StrongestLock a =
+  StrongestLock { getStrongestLock :: Maybe a}
+
+class ComparableLockStrength a where
+  -- | Create a lock that is at least as strong as both locks and at most as
+  -- strong as the stronger lock
+  -- Should satisfy:
+  --    If compare a b = GT
+  --        compareLockStrength (unionLocks a b) b = GT
+  --        compareLockStrength a (unionLocks a b) = EQ
+  --    If compare a b = EQ
+  --        compareLockStrength (unionLocks a b) b = EQ
+  --        compareLockStrength a (unionLocks a b) = EQ
+  --    If compare a b = LT
+  --        compareLockStrength (unionLocks a b) b = EQ
+  --        compareLockStrength a (unionLocks a b) = LT
+  unionLocks :: ComparableLockStrength a => a -> a -> a
+  compareLockStrength :: ComparableLockStrength a => a -> a -> Ordering
```

Comment:
If `compareLockStrength` is intended to be an `Ord`like method, we could do a default signature:

```suggestion
  unionLocks :: a -> a -> a
  compareLockStrength :: a -> a -> Ordering
  default compareLockStrength :: Ord a => a -> a -> Ordering
  compareLockStrength = compare
```

Reply:
I don't think we want the `Ord` constraint though since we can compare the locking strength of two locks where it doesn't make sense to have an `Ord` instance -> that is it only compare the `postgresRowLevelLockStrength` of `PostgresLockingKind` and not any of the other behavior.